src/main/java/DurationBeforeOrAfterCourseCompletion.java

```diff
@@ -0,0 +1,32 @@
+import java.util.Objects;
+
+public class DurationBeforeOrAfterCourseCompletion {
```

Comment:
would you consider to use Lombok or AutoValue libraries?

Reply:
I haven't come across this AutoValue library yet.
From Lombok I know only @Value library, but it's not appropriate here.